### PR TITLE
Add index on object_key field (needed when deleting assets).

### PR DESCRIPTION
**Pull request by rute-santos**
### Your pull request should…

* [x] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

**Comment by lkiesow** [comment]
This needs database migration scripts or does it automatically change the structure.
If it does not, can we? :D

**Comment by gregorydlogan** [comment]
Watch the outcome of https://github.com/opencast/opencast/pull/4179 for more details!

**Comment by rute-santos** [comment]
I've added the db migration scripts, but I'm not sure if I put them in the right place. Please let me know and I will change if necessary. Thanks!

**Comment by lkiesow** [comment]
Since it does require a database migration, this should go into `develop`.

File: docs/upgrade/11.11_to_11.12/mariadb.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+CREATE INDEX IF NOT EXISTS IX_oc_aws_asset_mapping_object_key ON oc_aws_asset_mapping (object_key);
```
Comment (mliradelc) [inline docs/upgrade/11.11_to_11.12/mariadb.sql]: Hi rute, thanks for your PR, now that is aiming to develop, can you change the folder name to `12_to_13` for both scripts. Thanks

**Review by mliradelc — CHANGES_REQUESTED** [review]

File: docs/upgrade/11.11_to_11.12/mariadb.sql (hunk 1 above)
Comment (rute-santos) [reply docs/upgrade/11.11_to_11.12/mariadb.sql]: Oops! Fixing...

**Pull request merged**